astropy/modeling/projections.py

```diff
@@ -158,6 +158,17 @@ class should override this method.
             self._prj.pv = None, *pv
             self._prj.set()
 
+    # Note that the Prjprm object can't be pickled, but we don't actually
+    # need to pickle it as it is recreated during __init__ and can be
+    # updated with update_prj. So instead we represent the state with a
+    # simple dictionary of parameter values.
+
+    def __getstate__(self):
+        return {p: getattr(self, p).value for p in self.param_names}
+
+    def __setstate__(self, state):
+        self.__init__(**state)
+
```

Comment:
These changes seem reasonable to make sure something pickles. I want @mcara (who made the most recent major changes) to comment  on this.

Also, @astrofrog do you know how far back this issue extends? it looks like the latest major changes to this made it into astropy 5.1. I wonder if the issue is present in versions earlier than that...

Reply:
I'm not sure if this ever worked, but can check later